mbedtls/src/cipher/mod.rs

```diff
@@ -290,12 +290,12 @@ impl Cipher<Encryption, Authenticated, AdditionalData> {
     pub fn encrypt_auth_inplace(
         mut self,
         ad: &[u8],
-        data: &mut [u8],
-        tag: &mut [u8],
+        data_with_tag: &mut [u8],
```

Comment:
This merits a doc comment explaining the order of data and tag in the input slice